internal/document.go

```diff
@@ -20,6 +22,25 @@ type Document struct {
 	Nonce     []byte `cbor:"nonce" json:"nonce,omitempty"`
 }
 
+func MakeDocumentFromBytes(bytes []byte) (Document, error) {
```

Comment:
Updated document to be closer to `CoseSign1` where we have a make-from-bytes function that is responsible for unmarshaling and verifying the correctness of fields---where correctness simply means required fields exist and that things are not too short/long.

Reply:
My recommendation would be to abandon `MakeDocumentFromBytes` (here and in `CoseSign1`) and make `Documnet` and `CoseSign1` to conform to the https://pkg.go.dev/encoding#BinaryMarshaler interface. You can do that by implementing `MarshalBinary` and `UnmarshalBinary` like:
```go
func (doc *Document) UnmarshalBinary(data []byte) error {
	return cbor.Unmarshal(data, doc)
}
```

Doing that could let you use `doc.MarshalBinary` in tests. You could also have then a `Verify` function on `Document` that would do all the verification. I'm a little concerned about `internal.Verify` needing to know a lot about how to sequence the different document verification steps - it would be easier to have them wrapped inside `Verify`. You could then implement a `Document.Certificates` functions to get the certificates for `coseSign1.CheckSignature`, but also a user getting a `Document` back (instead of a `Result`) could call `Document.Certificates`.

Reply:
Both Document and CoseSign1 implement `UnmarshalBinary` now. Removed the make functions and added verification functions